python/ray/serve/_private/common.py

```diff
@@ -353,11 +364,18 @@ def from_str(cls, actor_name):
         # can try to keep the internal name always hard coded with the prefix.
         replica_name = actor_name.replace(cls.prefix, "")
         parsed = replica_name.split(cls.delimiter)
-        assert len(parsed) == 2, (
-            f"Given replica name {replica_name} didn't match pattern, please "
-            f"ensure it has exactly two fields with delimiter {cls.delimiter}"
-        )
-        return cls(deployment_tag=parsed[0], replica_suffix=parsed[1])
+        if len(parsed) == 3:
+            return cls(
+                app_name=parsed[0], deployment_name=parsed[1], replica_suffix=parsed[2]
+            )
+        elif len(parsed) == 2:
+            return cls("", deployment_name=parsed[0], replica_suffix=parsed[1])
```

Comment:
I thought we're using "default" as default app name, aren't we?

Reply:
Unfortunately we still have to maintain compatibility with 1.x deployments which are deprecated but not yet removed. 1.x deployments don't have a concept of applications, so I use `app_name=""` to mark the 1.x deployments.